### ISSUE TITLE
Track project while disconnected

### ISSUE THREAD

**Issue by kim** [radicle-dev/radicle-link#141]
Currently, `git::repo::clone` requires a `PeerId`, and an established connection to that peer. It should also be possible to schedule a known identity hash to be cloned later, for example while offline.

Practically, this would create an empty repository (or namespace), which would technically be invalid, but the fetch/clone machinery should be able to detect and handle that case. It may also entail defining a startup-sequence for the networking stack, which would emit `want` and `have` gossip messages as appropriate (queued in secondary storage, or read from the repository state).

**Comment by FintanH** [comment]
I started thinking about the initial part of this about creating the empty state.

If we're offline and only have access to the storage and we're trying to track `<project_id>` from `<remote>` we could do the following:
* Create `refs/namespaces/<project_id>/refs/rad/self`
* Create `refs/namesapces/<project_id>/refs/remotes/<remote>/rad/id`

The above means that:
a) We have linked our `rad/self` to the project's `rad/self`
b) Won't have a `rad/id` ourselves for the project
c) The `rad/id` of the remote is empty

The idea being that we uniquely identify this case, and when we're online we can query the `RadUrn` + `PeerId` and replace the remote's `rad/id` with the actual info, and also finish the setup on our own local references.

In a way, it seems a bit hacky to have this "special case". But I'm not sure how to create the references otherwise without some `Oid`. 

As I'm typing this I'm wondering if maybe `Oid::zero` would work? :thinking:
